README.md

```diff
@@ -735,6 +723,56 @@ client.writeFile(recordType, readmeFile, null, new ResultHandler<RecordMeta>() {
 );
 ```
 
+Similarly, to read a file, use the `readFile` method. The `File` argument should be the destination that the plaintext
+file will be written to. If the file already exists, it will be overwritten. Continuing the example above, you
+could read the file written as follows:
+
+```
+...
+UUID readmeRecordId = ...; // Record ID of file written previously
+File destinationFile = new File("README-2.md");
+
+client.readFile(readmeRecordId, destinationFile, new ResultHandler<RecordMeta>() {
+    @Override
+    public void handle(Result<RecordMeta> r) {
+      if(r.isError()) {
+        // an error occurred
+        throw new RuntimeException(r.asError().other());
+      }
+
+      // `README-2.md` will contain contents of file, which we write to standard out.
+      System.out.println("README-2.md: " + new String(Files.readAllBytes(destinationFile), "UFT-8"));
+    }
+  }
+);
+```
+
+### Storage Requirements
+When uploading a file, the SDK expects to be able to (temporarily) store an encrypted version of the plaintext
+file in the same directory. Once the upload finishes (with or without error), the temporary file will be deleted.
```

Comment:
Do we want the encrypted file to be deleted if the server fails? Would this propagate up to the application, and it would then delete the plaintext data? How does the application know to re-try upload? Could this result in data loss?

Reply:
The application gives a callback to the SDK which is invoked only when the server returns a result or times out. Regardless of what that result is (success, timeout, "im a teapot", ...), the application will know something happened and can inspect the result to determine what to do.